### PR TITLE
Implement text string sets.

### PR DESCRIPTION
**Pull request by wxsBSD**
Add support for text string sets into the grammar. They look like this:

for any s in ("a", "b"): (pe.imphash() == s)

This requires changing integer_set and integer_enumeration to just be set and enumeration, and adding a new type (YR_ENUMERATION) that tracks the type of enumeration (integer or otherwise) and the number of items in the enumeration.

The enumeration now checks that each type of the item in the enumeration is consistent and will cause a compiler error if they are not. For example, this is an error:

for any s in ("a", 0): (s)

Also, fix the build when using --with-debug-verbose option as it was missing the assert.h include.

### CODE REVIEW

**Comment by wxsBSD** [comment]
Closes #1765.

**Comment by wxsBSD** [comment]
Forgot to mention that I'll update docs before merging if this looks good.

File: libyara/include/yara/exec.h [hunk 1]
```diff
@@ -123,6 +123,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define OP_OF_FOUND_IN           75
 #define OP_COUNT_IN              76
 #define OP_DEFINED               77
+#define OP_ITER_START_TEXT_STRING_SET 78
```
Comment (plusvic) [inline libyara/include/yara/exec.h:126]: nit: Re-align constants in the lines above with the `78` in this line.

**Comment by plusvic** [comment]
Looks good to me.

**Review by plusvic — APPROVED** [review]

**Comment by plusvic** [comment]
Waiting for doc update before merging.

**Pull request merged**